XpyriMentor/suggestors/sequential_strategizer.py

```diff
@@ -0,0 +1,109 @@
+        Parameters
+        ----------
+        * suggestors [`list[tuple[int, Suggestor]]`]:
+            A list of tuples where the first element is the number of suggestions the
+            suggestor can make and the second element is the suggestor. A negative number
+            of suggestions is interpreted as infinity, aand can only be used as the last
+            budget.
+
+            If the first suggestor is a DefaultSuggestor, it will be replaced with a
+            LHSSuggestor with the same budget. If any other suggestor is a
+            DefaultSuggestor, it will be replaced with a POSuggestor.
+        """
+        for n, (budget, suggestor) in enumerate(suggestors):
+            if isinstance(suggestor, DefaultSuggestor):
+                if n == 0:
+                    suggestors[n] = (
+                        budget,
+                        LHSSuggestor(
+                            space=suggestor.space, rng=suggestor.rng, n_points=budget
+                        )
+                    )
+                else:
+                    suggestors[n] = (
+                        budget,
+                        POSuggestor(
+                            space=suggestor.space,
+                            rng=suggestor.rng,
+                            n_objectives=suggestor.n_objectives,
+                        ))
+            if budget <= 0:
+                # Interpret negative budgets as infinity
+                suggestors[n] = (float("inf"), suggestors[n][1])
+            if hasattr(suggestors[n][1], "n_points"):
+                if suggestors[n][1].n_points != suggestors[n][0]:
+                    warnings.warn(
+                        f"Budget of {suggestors[n][0]} points does not match number of "
+                        "points for suggestor of type "
+                        f"{suggestors[n][1].__class__.__name__}."
+                    )
+            if math.isinf(suggestors[n][0]):
+                if n < len(suggestors) - 1:
+                    raise ValueError("Infinite budget must be the last budget")
+        self.suggestors = suggestors
+
+    def suggest(self, Xi: Iterable[Iterable], Yi: Iterable, n_asked: int = 1):
+        number_to_skip = len(Xi)
+        number_left_to_find = n_asked
+        suggestions = []
+        for budget, suggestor in self.suggestors:
+            if number_left_to_find == 0:
+                # If we have already found all the points we need, we can stop.
+                break
+            if number_to_skip >= budget:
+                # If we have been told enough points, we have to skip this suggestor.
+                number_to_skip -= budget
+                continue
+            if number_to_skip + number_left_to_find >= budget:
+                # If we need more points than the suggestor can give us, we take all the
+                # points the suggestor can give us and continue with the next suggestor.
+                number_from_this_suggestor = budget - number_to_skip
```

Comment:
Something seems a bit off here. I will take a deeper look later. 

What if you have 3 suggestors, 5 points  budget each and load in 7 data points.
How would that work. I guess it is the tracking of what has already happened and what budget each suggestor has that is a bit confusing.

Reply:
I'll move the testing whether we are done to a separate if clause at the end, and moved updating suggestions and `number_left_to_find` out of the main `if` loop. That makes the flow slightly clearer.

If we ask for 4 points, we will start out with `number_left_to_skip = 7` and `number_left_to_find =4 `

For the first suggestor, we have to skip it completely (`number_left_to_skip >= budget`; `7>=5`)

We then have `number_left_to_skip = 2` and `number_left_to_find =4 `

For the second suggestor, we will use it, but need more points than the suggestor can provide (`number_left_to_skip + number_left_to_find >= budget`; `2 + 4 > 5`). We use that suggestor to find 3 points (`number_from_this_suggestor = budget - number_left_to_skip = 5 - 2 = 3`).

We then have `number_left_to_skip = 0` and `number_left_to_find =1 `

We use the final suggestor to find one point (`else`).